thirdparty/cmdconfig/commands/cmdeval/cmdeval.go

```diff
@@ -205,29 +205,34 @@ func (r *EvalFnRunner) NewFunction() *kptfile.Function {
 
 // Add the evaluated function to the kptfile.Function list, this Function can either be
 // `pipeline.mutators` or `pipeline.validators`
-func (r *EvalFnRunner) updateFnList(oldFNs []kptfile.Function) []kptfile.Function {
+func (r *EvalFnRunner) updateFnList(oldFNs []kptfile.Function) ([]kptfile.Function, string) {
 	var newFns []kptfile.Function
-	pr := printer.FromContextOrDie(r.Ctx)
 	found := false
+	newFn := r.NewFunction()
+	var message string
 	for _, m := range oldFNs {
 		switch {
 		case m.Image != "" && m.Image == r.Image:
-			pr.Printf("found image in Kptfile, updating...\n")
-			newFns = append(newFns, *r.NewFunction())
+			newFns = append(newFns, *newFn)
 			found = true
+			message = fmt.Sprintf("Updated %v as %v in the Kptfile.", r.Image, r.FnType)
 		case m.Exec != "" && m.Exec == r.Exec:
-			pr.Printf("found exec in Kptfile, updating...\n")
-			newFns = append(newFns, *r.NewFunction())
+			newFns = append(newFns, *newFn)
 			found = true
+			message = fmt.Sprintf("Updated %v as %v in the Kptfile.", r.Exec, r.FnType)
 		default:
 			newFns = append(newFns, m)
 		}
 	}
 	if !found {
-		newFns = append(newFns, *r.NewFunction())
-		pr.Printf("adding function to Kptfile\n")
+		newFns = append(newFns, *newFn)
+		if newFn.Image != "" {
+			message = fmt.Sprintf("Added %v as %v in the Kptfile.", r.Image, r.FnType)
+		} else if newFn.Exec != "" {
+			message = fmt.Sprintf("Added %v as %v in the Kptfile.", r.Exec, r.FnType)
+		}
 	}
-	return newFns
+	return newFns, message
```

Comment:
we should use %q instead of %v for user provided input, but feel free to do that in the follow up PR.

it will read:

Updated "gcr.io/kpt-fn/set-namespace:v0.1.3" as mutator in the Kptfile.

Reply:
done.